impeller/geometry/color.cc

```diff
@@ -13,6 +13,28 @@
 
 namespace impeller {
 
+#define _IMPELLER_ASSERT_BLEND_MODE(blend_mode)                            \
+  auto enum_##blend_mode = static_cast<std::underlying_type_t<BlendMode>>( \
+      BlendMode::k##blend_mode);                                           \
+  if (i != enum_##blend_mode) {                                            \
```

Comment:
I'd turn `i` into a parameter so that the macro isn't making assumptions about the context it's used in.

Reply:
Hoping to keep this param out of the macro input of `IMPELLER_FOR_EACH_BLEND_MODE` if that's cool, since it shouldn't be needed for actual usage. Open to alternative ideas for verifying the macro matches the enum.

Reply:
It's not a huge deal but here's how you could do it without macros that implicitly capture their context.  We don't actually care about the asserted value, just that they are all accounted for, they all exist and there is no duplicates:

```c++
#define _IMPELLER_ASSERT_BLEND_MODE(blend_mode) \
  auto enum_##blend_mode = static_cast<std::underlying_type_t<BlendMode>>( \
      BlendMode::k##blend_mode);                                           \
  (void)enum_##blend_mode;
  
#define _IMPELLER_PLUS_ONE(blend_mode) + 1

static constexpr inline bool ValidateBlendModes() {
  IMPELLER_FOR_EACH_BLEND_MODE(_IMPELLER_ASSERT_BLEND_MODE);
  int tally = 0  IMPELLER_FOR_EACH_BLEND_MODE(_IMPELLER_PLUS_ONE);
  if (tally - 1 !=
      static_cast<std::underlying_type_t<BlendMode>>(BlendMode::kLast)) {
    return false;
  }
  return true;
}
```

Reply:
This is close, but it doesn't quite prevent duplicates. If kSrcOver was accidentally replaced with a second kSrc (thereby dropping kSrcOver), I think this check would pass?

Reply:
nope, you'd get a redefinition error when compiling, `enum_SrcOver` would be declared twice.

Reply:
ah, nice